share/module_bc.F

```diff
@@ -927,13 +928,27 @@ SUBROUTINE set_physical_bc3d( dat, variable_in,        &
 
 !  same procedure in y
 
+!  Set the starting and ending loop indexes in the 'i' direction, so that
+!  halo cells on the edge of the domain are also updated.  Begin with a default
+!  start and end index for inner tiles, and then modify if the tile is on the
+!  edge of the domain.
+
+      i_start = MAX(ids, its-1)
+      i_end = MIN(ite+1, ide+istag)
+      IF ( its .eq. ids) THEN
+        i_start = ims
```

Comment:
I wonder about ims here, because ims includes the halo size. Is this deliberate? Can ips be used?

Reply:
I took this straight from set_physical_bc2d for consistency. But yes it shouldn't be necessary to include the complete halo. We could also write i_start = its -1, i_end = ite+1 (so without the max/min and if statements). Then all cases should be included as well. Is that what you had in mind?